src/OmniSharp.Roslyn.CSharp/Services/Refactoring/RunFixAllCodeActionService.cs

```diff
@@ -0,0 +1,144 @@
+    public class RunFixAllCodeActionService : FixAllCodeActionBase, IRequestHandler<RunFixAllRequest, RunFixAllResponse>
+    {
+        private readonly ILogger<RunFixAllCodeActionService> _logger;
+        private readonly FixAllDiagnosticProvider _fixAllDiagnosticProvider;
+
+        [ImportingConstructor]
+        public RunFixAllCodeActionService(ICsDiagnosticWorker diagnosticWorker, CachingCodeFixProviderForProjects codeFixProvider, OmniSharpWorkspace workspace, ILoggerFactory loggerFactory) : base(diagnosticWorker, codeFixProvider, workspace)
+        {
+            _logger = loggerFactory.CreateLogger<RunFixAllCodeActionService>();
+            _fixAllDiagnosticProvider = new FixAllDiagnosticProvider(diagnosticWorker);
+        }
+
+        public async Task<RunFixAllResponse> Handle(RunFixAllRequest request)
+        {
+            if(request.Scope != FixAllScope.Document && request.FixAllFilter == null)
+                throw new NotImplementedException($"Only scope '{nameof(FixAllScope.Document)}' is currently supported when filter '{nameof(request.FixAllFilter)}' is not set.");
+
+            var solutionBeforeChanges = Workspace.CurrentSolution;
+
+            var mappedProvidersWithDiagnostics = await GetDiagnosticsMappedWithFixAllProviders(request.Scope, request.FileName);
+
+            var filteredProvidersWithFix = mappedProvidersWithDiagnostics
+                .Where(diagWithFix =>
+                {
+                    if (request.FixAllFilter == null)
+                        return true;
+
+                    return request.FixAllFilter.Any(x => diagWithFix.HasFixForId(x.Id));
+                });
+
+            foreach (var singleFixableProviderWithDocument in filteredProvidersWithFix)
+            {
+                try
+                {
+                    var document = Workspace.CurrentSolution.GetDocument(singleFixableProviderWithDocument.DocumentId);
+
+                    var fixer = singleFixableProviderWithDocument.FixAllProvider;
+
+                    var (action, fixableDiagnosticIds) = await singleFixableProviderWithDocument.RegisterCodeFixesOrDefault(document);
+
+                    if (action == null)
+                        continue;
+
+                    var fixAllContext = new FixAllContext(
+                        document,
+                        singleFixableProviderWithDocument.CodeFixProvider,
+                        Microsoft.CodeAnalysis.CodeFixes.FixAllScope.Project,
+                        action.EquivalenceKey,
+                        fixableDiagnosticIds,
+                        _fixAllDiagnosticProvider,
+                        CancellationToken.None
+                    );
+
+                    var fixes = await singleFixableProviderWithDocument.FixAllProvider.GetFixAsync(fixAllContext);
+
+                    if (fixes == null)
+                        continue;
+
+                    var operations = await fixes.GetOperationsAsync(CancellationToken.None);
```

Comment:
an approach with a request-supplied timeout would probably be better? https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Roslyn.CSharp/Services/Navigation/GotoDefinitionService.cs#L78

Reply:
Added timeout parameter